akka-testkit-typed/src/test/scala/akka/testkit/typed/scaladsl/BehaviorTestKitSpec.scala

```diff
@@ -203,7 +243,7 @@ class BehaviorTestKitSpec extends WordSpec with Matchers {
       testkit.run(SpawnAndWatchWith("hello"))
       val child = testkit.childInbox("hello").ref
       testkit.retrieveAllEffects() should be(Seq(
-        Spawned(Child.initial, "hello", Props.empty),
+        Spawned(Child.initial, "hello", Props.empty)(),
```

Comment:
I think these `()` looks strange as api. Perhaps we should craft these effect classes without `case class` and have friendly apply and unapply. You don't have to do that in this PR, but let's create a ticket.

Reply:
Or factory methods to match the java dsl

Reply:
done